### PR TITLE
Warn when using another Python implementation than CPython.

### PR DESCRIPTION
**Pull request by icemac**
PyPy cannot be restricted enough by RestrictedPython.
Other implementation might have other or similar issues.

### CODE REVIEW

File: src/RestrictedPython/compile.py [hunk 1]
```diff
@@ -19,9 +24,13 @@ def _compile_restricted_mode(
...
         policy=RestrictingNodeTransformer):
+
+    if not IS_CPYTHON:
+        warnings.warn(NOT_CPYTHON_WARNING, RuntimeWarning)
+
     byte_code = None
-    errors = []
-    warnings = []
+    collected_errors = []
+    collected_warnings = []
```
Comment (tseaver) [inline src/RestrictedPython/compile.py]: What does this refactoring have to do with the "not on PyPy" purpose of the PR?

File: docs/CHANGES.rst [hunk 2]
```diff
@@ -31,7 +32,7 @@ Changes
   `SelectCompiler.py`, `MutatingWorker.py`, `RestrictionMutator.py` and
   `tests/verify.py`.
 
-- Drop support of PyPy as there currently seems to be no way to restrict the
+- Drop support of PyPy as there currently is no way to restrict the
```
Comment (tseaver) [inline docs/CHANGES.rst]: "Drop support for PyPy" seems more idiomatic.

File: src/RestrictedPython/compile.py [hunk 3]
```diff
@@ -11,6 +12,10 @@
 syntax_error_template = (
     'Line {lineno}: {type}: {msg} in on statement: {statement}')
 
+NOT_CPYTHON_WARNING = (
+    'You are using on a not supported Python implementation.'
+    ' Use CPython to prevent security issues when using RestrictedPython!')
+
```
Comment (tseaver) [inline src/RestrictedPython/compile.py]: The message would read better as:

```python
NOT_CPYTHON_WARNING = (
    'RestrictedPython is only supported on CPython: use on other Python '
    'implementations may create security issues.'
)
```

**Comment by loechel** [comment]
The Tests for the warning seems not to work in PyPy. @icemac could you have another look.

File: src/RestrictedPython/compile.py (hunk 1 above)
Comment (icemac) [reply src/RestrictedPython/compile.py]: The `warnings` variable shadows the `warnings` module which is now needed some lines after the variable definition. I did not want to change only the name of one variable.

**Comment by icemac** [comment]
@loechel It worked on CPython because the warning was only emitted once. Are you okay with emitting a warning every time the function is used?

**Comment by icemac** [comment]
@tseaver Are you okay with my changes?

**Comment by loechel** [comment]
@icemac I am ok, with ``warnings.warn_explicte``, as that makes more sense, and helps if we want to try to make pypy in the long run possible.

I would like to have the explicite test keeped that runs on the can fail branch pypy in travis-ci.

**Pull request merged**